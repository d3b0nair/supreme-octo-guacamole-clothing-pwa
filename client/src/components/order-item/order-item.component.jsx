import React from "react";

import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectOrderData } from "../../redux/order/order.selectors";

import {
  OrderContainer,
  Title,
  ItemsPreview,
  CustomerInfo,
  OrderNumber,
  ItemContainer,
  ItemText,
  TotalPrice,
  InfoText,
  DeliveryInfo,
  DateText,
  InfoContainer
} from "./order-item.styles";

function OrderItem({ order_data }) {
  return (
    <OrderContainer onClick={(e) => e.stopPropagation()}>
      <OrderNumber>Order: {order_data.id}</OrderNumber>
      <Title>Delivery details:</Title>
      <InfoContainer>
      <CustomerInfo>
        <InfoText>Name: {order_data.customer.name}</InfoText>
        <InfoText>Email: {order_data.customer.email}</InfoText>
        <InfoText>Phone number: {order_data.customer.phone} </InfoText>
      </CustomerInfo>
      <DeliveryInfo>
        <InfoText>Street Adress: {order_data.customer.street}</InfoText>
        <InfoText>City: {order_data.customer.city}</InfoText>
        <InfoText>State: {order_data.customer.state} </InfoText>
        <InfoText>Zip Code: {order_data.customer.zip} </InfoText>
      </DeliveryInfo>
      </InfoContainer>
      <Title>Purchased Goods:</Title>
      <ItemsPreview>
        {order_data.items
          .map((item) => (
            <ItemContainer key={item.id}>
              <img alt="purchased product" src={item.imageUrl}/>
              <ItemText>{item.name}</ItemText>
              <ItemText>{item.price}$ x {item.quantity}</ItemText>
            </ItemContainer>
          ))}
      </ItemsPreview>
      <TotalPrice>TOTAL: ${order_data.total}</TotalPrice>
      <DateText>Purchased at {order_data.PlacedAt.slice(0, 24)}</DateText>
    </OrderContainer>
  );
}

const mapStateToProps = createStructuredSelector({
  order_data: selectOrderData,
});

export default connect(mapStateToProps)(OrderItem);
