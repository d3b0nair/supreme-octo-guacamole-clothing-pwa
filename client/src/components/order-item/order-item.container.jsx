import { connect } from "react-redux";
import { compose } from "redux";
import { createStructuredSelector } from "reselect";
import { selectIsSelectedOrderLoaded } from "../../redux/order/order.selectors";

import WithSpinner from "../with-spinner/with-spinner.component";
import OrderItem from "./order-item.component";

const mapStateToProps = createStructuredSelector({
  isLoading: (state) => !selectIsSelectedOrderLoaded(state),

});

const OrderContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(OrderItem);

export default OrderContainer;
