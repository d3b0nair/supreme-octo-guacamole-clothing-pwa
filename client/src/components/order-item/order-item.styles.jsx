import styled, { css } from "styled-components";

import { frostedGlass } from "./../../global.styles";

const borderOutline = css`
  border: 1px solid rgba(255, 255, 255, 1);
`;
const padding = css`
  padding: 20px;
  @media screen and (max-width: 800px) {
    padding: 15px;
  }
`;

const lineHeight = css`
  line-height: 55%;
  @media screen and (max-width: 800px) {
    line-height: 100%;
  }
`;

export const OrderContainer = styled.div`
  display: flex;
  position: relative;
  flex-direction: column;
  margin: auto;
  width: 700px;
  color: white;
  border-radius: 4px;
  min-width: 350px;
  ${frostedGlass}
  ${padding}
  ${borderOutline}

  @media screen and (max-width: 800px) {
    top: unset;
    transform: unset;
    left: unset;
    min-width: unset;
    width: 100%;
  }
`;

export const ItemContainer = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr;
  gap: 0px 0px;
  grid-template-areas: ". . . . .";
  min-height: 50px;
  align-items: center;
`;

export const Title = styled.h2`
  ${padding}
  text-align: center;
  font-size: 1.5em;
  ${lineHeight}
  @media screen and (max-width: 800px) {
    margin-top: 10%;
  }
`;
export const ItemText = styled.div`
  ${padding}
  text-align: center;
  font-size: 1.2em;
`;

export const TotalPrice = styled.h4`
  margin: 30px 0 0 auto;
  font-size: 2em;
  ${padding}
  ${lineHeight}
`;

export const ItemsPreview = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  border: 1px solid rgba(255, 255, 255, 1);
  ${frostedGlass}
  ${padding}
  img {
    width: 60px;
    margin: 5px;
    max-width: 100%;
    height: auto;
    border: 1px solid rgba(255, 255, 255, 1);
  }
  @media screen and (max-width: 800px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
    img {
      width: 50px;
      margin: 3px;
    }
  }
`;
export const CustomerInfo = styled.div`
  justify-self: left;
  @media screen and (max-width: 800px) {
    justify-self: unset;
  }
`;

export const DeliveryInfo = styled.div`
  justify-self: right;
  @media screen and (max-width: 800px) {
    justify-self: unset;
  }
`;

export const InfoContainer = styled.div`
  width: 100%;
  display: grid;
  border: 1px solid rgba(255, 255, 255, 1);
  ${frostedGlass}
  ${padding}
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr;
  align-items: center;
  @media screen and (max-width: 800px) {
    grid-template-columns: 1fr;
  }
`;

export const InfoText = styled.p`
  font-size: 1.2em;
  ${lineHeight}
`;

export const OrderNumber = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  color: #1b2735;
  ${frostedGlass}
  background: white;
  padding: 5px;
  border-radius: 0 7px 0 10px;
`;

export const DateText = styled.p`
  text-align: center;
  color: #1b2735;
  ${padding}
  ${lineHeight}
`;
