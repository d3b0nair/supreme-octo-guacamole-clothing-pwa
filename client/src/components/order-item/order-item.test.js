import React from "react";
import {
  render,
  cleanup,
} from "@testing-library/react";
import { OrderItem } from "./order-item.component";

describe("OrderItem component", () => {
  const mockItems = [];
  const mockCustomer = {
    name: "",
    email: "",
    phone: "",
    street: "",
    city: "",
    state: "",
    zip: "",
  };

  const mockProps = (otherProps) => ({
    ...otherProps,
    order_data: {
      total: "",
      customer: mockCustomer,
      items: mockItems,
      PlacedAt: "",
    },
  });
  const component = (mockProps) => <OrderItem {...mockProps} />;

  afterEach(cleanup);

  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
});
