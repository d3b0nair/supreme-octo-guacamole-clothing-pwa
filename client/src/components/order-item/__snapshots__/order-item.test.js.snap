// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem component should mount 1`] = `
<div>
  <div
    class="sc-bdvvtL joPSmO"
  >
    <div
      class="sc-jrQzAO bmzRBB"
    >
      Order: 
    </div>
    <h2
      class="sc-dkPtRN jjcHws"
    >
      Delivery details:
    </h2>
    <div
      class="sc-furwcr gnIKgj"
    >
      <div
        class="sc-gKclnd cCFUfP"
      >
        <p
          class="sc-pVTFL ggGDnf"
        >
          Name: 
          
        </p>
        <p
          class="sc-pVTFL ggGDnf"
        >
          Email: 
          
        </p>
        <p
          class="sc-pVTFL ggGDnf"
        >
          Phone number: 
          
           
        </p>
      </div>
      <div
        class="sc-iCfMLu bXrODd"
      >
        <p
          class="sc-pVTFL ggGDnf"
        >
          Street Adress: 
          
        </p>
        <p
          class="sc-pVTFL ggGDnf"
        >
          City: 
          
        </p>
        <p
          class="sc-pVTFL ggGDnf"
        >
          State: 
          
           
        </p>
        <p
          class="sc-pVTFL ggGDnf"
        >
          Zip Code: 
          
           
        </p>
      </div>
    </div>
    <h2
      class="sc-dkPtRN jjcHws"
    >
      Purchased Goods:
    </h2>
    <div
      class="sc-jRQBWg dcFORt"
    />
    <h4
      class="sc-eCImPb ecssnl"
    >
      TOTAL: $
      
    </h4>
    <p
      class="sc-kDTinF ooKlQ"
    >
      Purchased at 
      
    </p>
  </div>
</div>
`;
